### PR TITLE
Allow Allfaces drawtypes to have 6 textures

### PR DESCRIPTION
**Pull request by DragonWrangler1**
- Goal of the PR
Allow for allfaces drawtypes to have 6 textures
- How does the PR work?
It adds merely changes to the allfaces drawtype registry to allow it have 6 textures instead of one
- Does it resolve any reported issue?
Yes: 
https://github.com/minetest/minetest/issues/9210
https://github.com/minetest/minetest/issues/6929

Of which #9210 was marked concept approved.

## To do

Ready for Review.

## How to test

Compile, make an allfaces node following the documentation to layout the textures, and ensure it works without side-effects

### CODE REVIEW

**Comment by DragonWrangler1** [comment]
Now I do have a question. Is it wanted for me to make an allfaces6 draw type, or just leave it to updating it.

**Comment by appgurueu** [comment]
> Now I do have a question. Is it wanted for me to make an allfaces6 draw type, or just leave it to updating the existing allfaces one.

I don't see a problem with reusing the existing drawtype as long as nodes with `tiles = {"blah.png"}` continue to work. If a modder expects some textures to be ignored (as in `tiles = {"blah.png", "blub.png"}`) that's on them.

**Comment by DragonWrangler1** [comment]
> I don't see a problem with reusing the existing drawtype as long as nodes with tiles = {"blah.png"} continue to work. If a modder expects some textures to be ignored (as in tiles = {"blah.png", "blub.png"}) that's on them.

From my test old allfaces blocks still work with just one texture.

File: doc/lua_api.md [hunk 1]
```diff
@@ -1471,6 +1471,11 @@ Look for examples in `games/devtest` or `games/minetest_game`.
 * `allfaces`
     * Often used for partially-transparent nodes.
     * External and internal sides of textures are visible.
+    * For This drawtype textures are as follows in the exact order:
+    ```
+        tiles = {tile definition 1, def2, def3, def4, def5, def6},
+        Textures of node;       -Y,  +Z,   +X,   -Z,   +Y,   -X
```
Comment (sfan5) [inline doc/lua_api.md:1477]: is this different from normal nodes?
if yes: why?
if no: we shouldn't duplicate it here

File: doc/lua_api.md [hunk 2]
```diff
@@ -1470,7 +1470,7 @@ Look for examples in `games/devtest` or `games/minetest_game`.
       'Connected Glass'.
 * `allfaces`
     * Often used for partially-transparent nodes.
-    * External and internal sides of textures are visible.
+    * External sides of textures are visible.
```
Comment (appgurueu) [inline doc/lua_api.md:1473]: Was the documentation wrong before or why this change?

Comment (DragonWrangler1) [reply doc/lua_api.md:1473]: It never showed the internal textures of the block from the inside, only those of other blocks.

**Comment by DragonWrangler1** [comment]
@sfence done.

File: src/client/content_mapblock.cpp [hunk 3]
```diff
@@ -1001,8 +1001,20 @@ void MapblockMeshGenerator::drawGlasslikeFramedNode()
...
-	useTile(0, 0, 0);
-	drawAutoLightedCuboid(box);
+	static const v3s16 tile_dirs[6] = {
+		v3s16(0, 1, 0),
+		v3s16(0, -1, 0),
+		v3s16(1, 0, 0),
+		v3s16(-1, 0, 0),
+		v3s16(0, 0, 1),
+		v3s16(0, 0, -1)
+	};
```
Comment (SmallJoker) [inline src/client/content_mapblock.cpp:1011]: Duplicate of `nodebox_tile_dirs`

Comment (DragonWrangler1) [reply src/client/content_mapblock.cpp]: Fixed.

File: games/devtest/mods/testnodes/drawtypes.lua [hunk 4]
```diff
@@ -98,6 +98,16 @@ minetest.register_node("testnodes:allfaces", {
 	groups = { dig_immediate = 3 },
 })
 
+minetest.register_node("testnodes:allfaces_6", {
+	description = S("\"allfaces 6 Textures\" Drawtype Test Node").."\n"..
+		S("Transparent node with visible internal backfaces"),
+	drawtype = "allfaces",
+	paramtype = "light",
+	tiles = { "testnodes_allfaces.png^[colorize:red", "testnodes_allfaces.png^[colorize:orange", "testnodes_allfaces.png^[colorize:yellow", "testnodes_allfaces.png^[colorize:green", "testnodes_allfaces.png^[colorize:blue", "testnodes_allfaces.png^[colorize:purple" },
```
Comment (SmallJoker) [inline games/devtest/mods/testnodes/drawtypes.lua:106]: Please break the line to max. 80 characters. https://dev.minetest.net/Lua_code_style_guidelines
or 95 if you apply the C++ guidelines.... https://dev.minetest.net/Code_style_guidelines#Don.27t_let_things_get_too_large

**Review by SmallJoker — COMMENTED** [review]
Works. master: red node, PR: individual colors.

comparison

File: games/devtest/mods/testnodes/drawtypes.lua (hunk 4 above)
Comment (DragonWrangler1) [reply games/devtest/mods/testnodes/drawtypes.lua]: Should be resolved.

**Review by SmallJoker — APPROVED** [review]

File: doc/lua_api.md (hunk 2 above)
Comment (appgurueu) [reply doc/lua_api.md:1473]: "Internal" is meant as in "internal" faces between allfaces nodes aren't culled, not backfaces generally. See Krock's screenshot: 
Screenshot from 2024-09-20 15-02-53

I agree that the current wording is poor, but just removing it isn't right and would leave the user wondering what this drawtype does at all. After all external sides are visible for regular nodes just as well.

Comment (DragonWrangler1) [reply doc/lua_api.md]: @appgurueu I updated the wording
